extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -85,19 +86,36 @@ const FUNCTION_WHITELIST = (function() {
 })();
 
 /**
+ * Default maximum number of nodes in an expression AST.
+ * Double size of a "typical" expression in examples/bind/performance.amp.html.
+ * @const @private {number}
+ */
+const DEFAULT_MAX_AST_SIZE = 50;
+
+/**
  * A single Bind expression.
  */
 export class BindExpression {
   /**
    * @param {string} expressionString
+   * @param {number=} opt_maxAstSize
    * @throws {Error} On malformed expressions.
    */
-  constructor(expressionString) {
+  constructor(expressionString, opt_maxAstSize) {
     /** @const {string} */
     this.expressionString = expressionString;
 
     /** @const @private {!./bind-expr-defines.AstNode} */
     this.ast_ = parser.parse(this.expressionString);
+
+    // Check if this expression string is too large (for performance).
+    const size = this.numberOfNodesInAst_(this.ast_);
```

Comment:
Good question. Looks like a few ms in total on `examples/bind/performance.amp.html` with 5x throttling.